src/relations/backend_database.py

```diff
@@ -246,6 +245,16 @@ def collect_databases(self) -> list[str]:
 
         return databases
 
+    def generate_monitoring_hash(self, monitoring_password: str) -> str:
+        """Generate monitoring SCRAM hash against the current backend."""
+        conn = None
+        try:
+            with self.postgres._connect_to_database() as conn:
+                return get_scram_password(self.stats_user, monitoring_password, conn)
+        finally:
+            if conn:
+                conn.close()
+
```

Comment:
Generate SCRAM hash for the monitoring user, so that we can switch off md5 for the clients.